presto-main/src/main/java/com/facebook/presto/operator/scalar/CanonicalizePhoneNumberFunction.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.PhoneNumberUtil.PhoneNumberFormat;
+import com.google.i18n.phonenumbers.Phonenumber.PhoneNumber;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.util.Arrays;
+
+public class CanonicalizePhoneNumberFunction
+{
+    private static final PhoneNumberUtil PHONE_NUMBER_UTIL = PhoneNumberUtil.getInstance();
```

Comment:
This is thread-safe?

Reply:
I haven't been able to find official documentation on this - however because of the following I feel confident about this class being thread safe:

1) there are classes in the package that have the annotation - `<p>This class is not thread-safe.` example: `PhoneNumberMatcher`. This class does not have this annotation.

2) I read through the code for the functions that I am invoking and didn't find any write actions being performed on global data structures.